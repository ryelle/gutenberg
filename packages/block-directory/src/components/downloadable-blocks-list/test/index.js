/**
 * External dependencies
 */
import { render } from '@testing-library/react';

/**
 * WordPress dependencies
 */
import { useSelect } from '@wordpress/data';

/**
 * Internal dependencies
 */
import DownloadableBlocksList from '../';
import { items } from '../../test/fixtures';

jest.mock( '@wordpress/data/src/components/use-select', () => {
	// This allows us to tweak the returned value on each test
	const mock = jest.fn();
	return mock;
} );

jest.mock( '@wordpress/data/src/components/use-dispatch', () => ( {
	useDispatch: () => ( { installBlockType: jest.fn() } ),
} ) );

describe( 'DownloadableBlocksList', () => {
	describe( 'List rendering', () => {
		useSelect.mockImplementation( () => ( {
			isLoading: false,
			isInstallable: true,
		} ) );

		it( 'should render and empty list', () => {
			const { container } = render(
				<DownloadableBlocksList
					items={ [] }
					onSelect={ jest.fn() }
					onHover={ jest.fn() }
				/>
			);
			const downloadableBlocks = container.querySelector(
				'.block-directory-downloadable-blocks-list'
			);

			expect( downloadableBlocks ).toBe( null );
		} );

		it( 'should render plugins items into the list', () => {
			const { container } = render(
				<DownloadableBlocksList
					items={ items }
					onSelect={ jest.fn() }
					onHover={ jest.fn() }
				/>
			);
			expect( container ).toMatchSnapshot();
		} );
	} );
} );
