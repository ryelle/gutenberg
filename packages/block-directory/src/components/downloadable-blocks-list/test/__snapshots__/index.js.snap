// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DownloadableBlocksList List rendering should render plugins items into the list 1`] = `
<div>
  <div
    aria-label="todo"
    class="block-directory-downloadable-blocks-list"
    id="id-kulsib"
    role="listbox"
  >
    <div
      class="block-directory-downloadable-block-list-item"
    >
      <button
        class="components-button block-directory-downloadable-block-list-item__item"
        id="id-kulsib-1"
        role="option"
        tabindex="0"
        type="button"
      >
        <div
          class="block-directory-downloadable-block-icon"
        >
          <span
            class="block-editor-block-icon has-colors"
          >
            <svg
              aria-hidden="true"
              class="dashicon dashicons-block-default"
              focusable="false"
              height="20"
              role="img"
              viewBox="0 0 20 20"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M15 6V4h-3v2H8V4H5v2H4c-.6 0-1 .4-1 1v8h14V7c0-.6-.4-1-1-1h-1z"
              />
            </svg>
          </span>
        </div>
        <span
          class="block-directory-downloadable-block-list-item__details"
        >
          <span
            class="block-directory-downloadable-block-list-item__title"
          >
            Boxer 
            <span
              class="block-directory-downloadable-block-list-item__author"
            >
              by CK Lee
            </span>
          </span>
          <span
            class="block-directory-block-ratings"
          >
            <span
              aria-label="5 out of 5 stars"
            >
              <svg
                aria-hidden="true"
                class="block-directory-block-ratings__star-full"
                focusable="false"
                height="16"
                role="img"
                viewBox="0 0 24 24"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11.776 4.454a.25.25 0 01.448 0l2.069 4.192a.25.25 0 00.188.137l4.626.672a.25.25 0 01.139.426l-3.348 3.263a.25.25 0 00-.072.222l.79 4.607a.25.25 0 01-.362.263l-4.138-2.175a.25.25 0 00-.232 0l-4.138 2.175a.25.25 0 01-.363-.263l.79-4.607a.25.25 0 00-.071-.222L4.754 9.881a.25.25 0 01.139-.426l4.626-.672a.25.25 0 00.188-.137l2.069-4.192z"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="block-directory-block-ratings__star-full"
                focusable="false"
                height="16"
                role="img"
                viewBox="0 0 24 24"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11.776 4.454a.25.25 0 01.448 0l2.069 4.192a.25.25 0 00.188.137l4.626.672a.25.25 0 01.139.426l-3.348 3.263a.25.25 0 00-.072.222l.79 4.607a.25.25 0 01-.362.263l-4.138-2.175a.25.25 0 00-.232 0l-4.138 2.175a.25.25 0 01-.363-.263l.79-4.607a.25.25 0 00-.071-.222L4.754 9.881a.25.25 0 01.139-.426l4.626-.672a.25.25 0 00.188-.137l2.069-4.192z"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="block-directory-block-ratings__star-full"
                focusable="false"
                height="16"
                role="img"
                viewBox="0 0 24 24"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11.776 4.454a.25.25 0 01.448 0l2.069 4.192a.25.25 0 00.188.137l4.626.672a.25.25 0 01.139.426l-3.348 3.263a.25.25 0 00-.072.222l.79 4.607a.25.25 0 01-.362.263l-4.138-2.175a.25.25 0 00-.232 0l-4.138 2.175a.25.25 0 01-.363-.263l.79-4.607a.25.25 0 00-.071-.222L4.754 9.881a.25.25 0 01.139-.426l4.626-.672a.25.25 0 00.188-.137l2.069-4.192z"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="block-directory-block-ratings__star-full"
                focusable="false"
                height="16"
                role="img"
                viewBox="0 0 24 24"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11.776 4.454a.25.25 0 01.448 0l2.069 4.192a.25.25 0 00.188.137l4.626.672a.25.25 0 01.139.426l-3.348 3.263a.25.25 0 00-.072.222l.79 4.607a.25.25 0 01-.362.263l-4.138-2.175a.25.25 0 00-.232 0l-4.138 2.175a.25.25 0 01-.363-.263l.79-4.607a.25.25 0 00-.071-.222L4.754 9.881a.25.25 0 01.139-.426l4.626-.672a.25.25 0 00.188-.137l2.069-4.192z"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="block-directory-block-ratings__star-full"
                focusable="false"
                height="16"
                role="img"
                viewBox="0 0 24 24"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11.776 4.454a.25.25 0 01.448 0l2.069 4.192a.25.25 0 00.188.137l4.626.672a.25.25 0 01.139.426l-3.348 3.263a.25.25 0 00-.072.222l.79 4.607a.25.25 0 01-.362.263l-4.138-2.175a.25.25 0 00-.232 0l-4.138 2.175a.25.25 0 01-.363-.263l.79-4.607a.25.25 0 00-.071-.222L4.754 9.881a.25.25 0 01.139-.426l4.626-.672a.25.25 0 00.188-.137l2.069-4.192z"
                />
              </svg>
            </span>
            <span
              aria-label="%d total rating"
              class="block-directory-block-ratings__rating-count"
            >
              (
              )
            </span>
          </span>
        </span>
        <p
          class="block-directory-downloadable-block-list-item__desc"
        >
          Boxer is a Block that puts your WordPress posts into boxes on a page.
        </p>
      </button>
    </div>
    <div
      class="block-directory-downloadable-block-list-item"
    >
      <button
        class="components-button block-directory-downloadable-block-list-item__item"
        id="id-kulsib-2"
        role="option"
        tabindex="-1"
        type="button"
      >
        <div
          class="block-directory-downloadable-block-icon"
        >
          <span
            class="block-editor-block-icon has-colors"
          >
            <svg
              aria-hidden="true"
              class="dashicon dashicons-block-default"
              focusable="false"
              height="20"
              role="img"
              viewBox="0 0 20 20"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M15 6V4h-3v2H8V4H5v2H4c-.6 0-1 .4-1 1v8h14V7c0-.6-.4-1-1-1h-1z"
              />
            </svg>
          </span>
        </div>
        <span
          class="block-directory-downloadable-block-list-item__details"
        >
          <span
            class="block-directory-downloadable-block-list-item__title"
          >
            Boxer 
            <span
              class="block-directory-downloadable-block-list-item__author"
            >
              by CK Lee
            </span>
          </span>
          <span
            class="block-directory-block-ratings"
          >
            <span
              aria-label="5 out of 5 stars"
            >
              <svg
                aria-hidden="true"
                class="block-directory-block-ratings__star-full"
                focusable="false"
                height="16"
                role="img"
                viewBox="0 0 24 24"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11.776 4.454a.25.25 0 01.448 0l2.069 4.192a.25.25 0 00.188.137l4.626.672a.25.25 0 01.139.426l-3.348 3.263a.25.25 0 00-.072.222l.79 4.607a.25.25 0 01-.362.263l-4.138-2.175a.25.25 0 00-.232 0l-4.138 2.175a.25.25 0 01-.363-.263l.79-4.607a.25.25 0 00-.071-.222L4.754 9.881a.25.25 0 01.139-.426l4.626-.672a.25.25 0 00.188-.137l2.069-4.192z"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="block-directory-block-ratings__star-full"
                focusable="false"
                height="16"
                role="img"
                viewBox="0 0 24 24"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11.776 4.454a.25.25 0 01.448 0l2.069 4.192a.25.25 0 00.188.137l4.626.672a.25.25 0 01.139.426l-3.348 3.263a.25.25 0 00-.072.222l.79 4.607a.25.25 0 01-.362.263l-4.138-2.175a.25.25 0 00-.232 0l-4.138 2.175a.25.25 0 01-.363-.263l.79-4.607a.25.25 0 00-.071-.222L4.754 9.881a.25.25 0 01.139-.426l4.626-.672a.25.25 0 00.188-.137l2.069-4.192z"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="block-directory-block-ratings__star-full"
                focusable="false"
                height="16"
                role="img"
                viewBox="0 0 24 24"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11.776 4.454a.25.25 0 01.448 0l2.069 4.192a.25.25 0 00.188.137l4.626.672a.25.25 0 01.139.426l-3.348 3.263a.25.25 0 00-.072.222l.79 4.607a.25.25 0 01-.362.263l-4.138-2.175a.25.25 0 00-.232 0l-4.138 2.175a.25.25 0 01-.363-.263l.79-4.607a.25.25 0 00-.071-.222L4.754 9.881a.25.25 0 01.139-.426l4.626-.672a.25.25 0 00.188-.137l2.069-4.192z"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="block-directory-block-ratings__star-full"
                focusable="false"
                height="16"
                role="img"
                viewBox="0 0 24 24"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11.776 4.454a.25.25 0 01.448 0l2.069 4.192a.25.25 0 00.188.137l4.626.672a.25.25 0 01.139.426l-3.348 3.263a.25.25 0 00-.072.222l.79 4.607a.25.25 0 01-.362.263l-4.138-2.175a.25.25 0 00-.232 0l-4.138 2.175a.25.25 0 01-.363-.263l.79-4.607a.25.25 0 00-.071-.222L4.754 9.881a.25.25 0 01.139-.426l4.626-.672a.25.25 0 00.188-.137l2.069-4.192z"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="block-directory-block-ratings__star-full"
                focusable="false"
                height="16"
                role="img"
                viewBox="0 0 24 24"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11.776 4.454a.25.25 0 01.448 0l2.069 4.192a.25.25 0 00.188.137l4.626.672a.25.25 0 01.139.426l-3.348 3.263a.25.25 0 00-.072.222l.79 4.607a.25.25 0 01-.362.263l-4.138-2.175a.25.25 0 00-.232 0l-4.138 2.175a.25.25 0 01-.363-.263l.79-4.607a.25.25 0 00-.071-.222L4.754 9.881a.25.25 0 01.139-.426l4.626-.672a.25.25 0 00.188-.137l2.069-4.192z"
                />
              </svg>
            </span>
            <span
              aria-label="%d total rating"
              class="block-directory-block-ratings__rating-count"
            >
              (
              )
            </span>
          </span>
        </span>
        <p
          class="block-directory-downloadable-block-list-item__desc"
        >
          Boxer is a Block that puts your WordPress posts into boxes on a page.
        </p>
      </button>
    </div>
  </div>
</div>
`;
